### ISSUE TITLE
expectTypeOf - Deep nesting passes tests for different types

### ISSUE THREAD

**Issue by ivodolenc** [mmkal/expect-type#31]
Hi, I was just searching for some type testing lib and came across this nice project.

I did some quick nesting tests, so please correct me if I'm wrong.

I'm not sure if this is intentional, but here's an example to give you a better idea of what I mean:

### Passes

```ts
import { expectTypeOf } from 'expect-type'

type Test = {
  object: {
    object: {
      number: number
      string: string
      boolean: boolean
      object: {
        number: number
        string: string
        boolean: boolean
      }
    }
  }
}

type TestNotEqual = {
  object: {
    object: {
      number?: number
      object?: {
        number?: number
      }
    }
  }
}

expectTypeOf<Test>().toEqualTypeOf<TestNotEqual>() // this passes the test and does not report an error
```

### Not Passes

```ts
import { expectTypeOf } from 'expect-type'

type Test = {
  object: {
    string: string
    object: {
      object: {
        number: number
      }
    }
  }
}

type TestNotEqual = {
  object: {
    number: number // registers the number as an error since it should be a string, which is expected
    object: {
      object: {
        number: number
      }
    }
  }
}

expectTypeOf<Test>().toEqualTypeOf<TestNotEqual>() // this fails the test and throws an error
```

**Comment by mmkal** [comment]
Doesn't repro, likely fixed by #21 - @ivodolenc try v0.16.0

**Comment by ivodolenc** [comment]
Hey @mmkal, I just did a quick test, it works as expected in `v0.16.0` 👍
